### ISSUE TITLE
Pagination with null values crashes json parser

### ISSUE THREAD

**Issue by BenjHawk** [f3ath/json-api-dart#123]
As specified under https://jsonapi.org/format/#fetching-pagination 

_Keys MUST either be omitted_ **or have a null value** _to indicate that a particular link is unavailable._

Currently only variant 1 (ommiting the <Key, Value> pair) is working. A <Key, Value> pair where Value == null will crash the application when trying to parse.

The exception originates in 'inbound_document.dart'.

```
  Link _link(Object json) {
    //function body
  }
```

Solution proposal: change _link() function like

```
  Link _link(Object? json) {
    if(json == null)
      return null;
    //rest of function body
  }
```

and then parse to Map<String, Link?>

A sample of a response that will crash the application:

```
links": {
        "self": "https://<some-api-page>",
        "first": "https://<some-api-page>",
        "last": "https://<some-api-page>",
        "prev": null,
        "next": null
    },
```

**Comment by f3ath** [comment]
Thanks for reporting this. The issue should be fixed in 5.0.5. Null values are now excluded from the links object. Please confirm if it fixes the issue for you.

**Comment by BenjHawk** [comment]
Thank you very much for your quick reply @f3ath !

That fixed the problem for me.
👍
